src/main/java/gyro/google/storage/BucketResource.java

```diff
@@ -0,0 +1,635 @@
+    }
+
+    /**
+     * The bucket's logging configuration.
+     *
+     *  @subresource gyro.google.storage.BucketLogging
+     */
+    @Updatable
+    public BucketLogging getLogging() {
+        return logging;
+    }
+
+    public void setLogging(BucketLogging logging) {
+        this.logging = logging;
+    }
+
+    /**
+     * Minimum age an object in the bucket must reach before it can be deleted or overwritten. See also `Retention policies <https://cloud.google.com/storage/docs/bucket-lock?#retention-policy/>`_.
+     *
+     * @subresource gyro.google.storage.BucketRetentionPolicy
+     */
+    @Updatable
+    public BucketRetentionPolicy getRetentionPolicy() {
+        return retentionPolicy;
+    }
+
+    public void setRetentionPolicy(BucketRetentionPolicy retentionPolicy) {
+        this.retentionPolicy = retentionPolicy;
+    }
+
+    /**
+     * Bucket's default storage class used whenever no ``storageClass`` is specified for a newly-created object. Valid values are ``STANDARD``, ``NEARLINE``, ``COLDLINE``, ``MULTI-REGIONAL``, ``REGIONAL`` or ``DURABLE_REDUCED_AVAILABILITY``. Defaults to ``STANDARD``.
+     */
+    @Updatable
+    @ValidStrings({ "STANDARD", "NEARLINE", "COLDLINE", "MULTI-REGIONAL", "REGIONAL", "DURABLE_REDUCED_AVAILABILITY" })
+    public String getStorageClass() {
+        return storageClass;
+    }
+
+    public void setStorageClass(String storageClass) {
+        this.storageClass = storageClass;
+    }
+
+    /**
+     * The bucket's versioning configuration.
+     *
+     * @subresource gyro.google.storage. BucketVersioning
+     */
+    @Updatable
+    public BucketVersioning getVersioning() {
+        return versioning;
+    }
+
+    public void setVersioning(BucketVersioning versioning) {
+        this.versioning = versioning;
+    }
+
+    /**
+     * The bucket's website configuration controlling how the service behaves when accessing bucket contents as a web site.
+     *
+     * @subresource gyro.google.storage.BucketWebsite
+     */
+    @Updatable
+    public BucketWebsite getWebsite() {
+        return website;
+    }
+
+    public void setWebsite(BucketWebsite website) {
+        this.website = website;
+    }
+
+    /**
+     * The generated URI of this bucket.
+     */
+    @Output
+    public String getSelfLink() {
+        return selfLink;
+    }
+
+    public void setSelfLink(String selfLink) {
+        this.selfLink = selfLink;
+    }
+
+    @Override
+    public boolean doRefresh() throws Exception {
+        Storage storage = createClient(Storage.class);
+
+        try {
+            Bucket bucket = storage.buckets()
+                .get(getName())
+                .setProjection("full")
+                .execute();
+
+            if (bucket == null) {
+                return false;
+            }
+
+            copyFrom(bucket);
+
+            return true;
+        } catch (GoogleJsonResponseException e) {
+            if (e.getDetails().getCode() == 404) {
```

Comment:
This is caught by `GoogleResource.refresh()`.

Reply:
Fixed.